vertexai/src/GenerateContentResponse.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace Firebase.VertexAI {
+
+public readonly struct GenerateContentResponse {
+  public IEnumerable<Candidate> Candidates { get; }
+  public PromptFeedback? PromptFeedback { get; }
+  public UsageMetadata? UsageMetadata { get; }
+
+  // Helper properties
+  // The response's content as text, if it exists
+  public string Text { get; }
+
+  // Returns function calls found in any Parts of the first candidate of the response, if any.
+  public IEnumerable<ModelContent.FunctionCallPart> FunctionCalls { get; }
+}
+
+public enum BlockReason {
+  Unknown,
+  Safety,
+  Other,
+  Blocklist,
+  ProhibitedContent,
+}
+
+public readonly struct PromptFeedback {
+  public BlockReason? BlockReason { get; }
+  public string BlockReasonMessage { get; }
+  public IEnumerable<SafetyRating> SafetyRatings { get; }
+
+  // Hidden constructor, users don't need to make this
+}
+
+public readonly struct UsageMetadata {
+  public int PromptTokenCount { get; }
+  public int CandidatesTokenCount { get; }
+  public int TotalTokenCount { get; }
```

Comment:
not blocking, maybe consider to add related modality token info similar to this PR as well? https://github.com/firebase/flutterfire/pull/17065

Reply:
Yeah, I'll need to reorganize what my plans are for some of those based on the recent changes, but I'll wait until I'm doing the implementation of these parts, just to let the other platforms settle in on something for sure.